### PR TITLE
Fix when PartitionCount is configured as 0, the index of the array is illegal

### PR DESCRIPTION
**Pull request by CrazyHZM**
# Description

Fix when `PartitionCount` is configured as 0, the index of the array is illegal
Handling some exceptions
Fix typo

## Issue reference

Fix https://github.com/dapr/dapr/issues/3899

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: pkg/actors/actors.go [hunk 1]
```diff
@@ -904,7 +918,7 @@ func (m *ActorMetadata) removeReminderFromPartition(reminderRefs []actorReminder
 }
 
 func (m *ActorMetadata) insertReminderInPartition(reminderRefs []actorReminderReference, reminder *Reminder) ([]Reminder, actorReminderReference, string, *string) {
-	newReminderRef := m.createReminderReference(reminder)
+	newReminderRef, _ := m.createReminderReference(reminder)
```
Comment (daixiang0) [inline pkg/actors/actors.go:921]: Could we skip this error safely?

Comment (CrazyHZM) [reply pkg/actors/actors.go]: > Could we skip this error safely?

I think this cannot be ignored, it has been modified.

**Comment by yaron2** [comment]
Thanks @CrazyHZM. Can you please create an issue for this and reference it here?

/cc @artursouza

**Comment by CrazyHZM** [comment]
> Thanks @CrazyHZM. Can you please create an issue for this and reference it here?
> 
> /cc @artursouza

Done.

**Comment by daixiang0** [comment]
Could we have a ut for this case?

**Comment by daixiang0** [comment]
@CrazyHZM please fix DCO issue.

**Comment by CrazyHZM** [comment]
@daixiang0 
Because there are several commits that take too long, the rebase operation to solve the singoff will cause the PR to be not clean enough, I re-submitted the PR:
https://github.com/dapr/dapr/pull/4100

**Pull request closed without merging**